code/src/java/pcgen/cdom/facet/LevelInfoFacet.java

```diff
@@ -58,14 +59,14 @@
 	 * @return The object in this LevelInfoFacet for the Player Character
 	 *         represented by the given CharID and location.
 	 */
-	public PCLevelInfo get(CharID id, int location)
+	public Optional<PCLevelInfo> get(CharID id, int location)
```

Comment:
This implementation leaves something to be desired.  It returns an Optional, telegraphing semantically that something is optional, only to have all uses immediately call "get()".  This makes no sense.  If it's required, then the return null below should throw an exception.

Having said that, this change is also not well implemented because some of the gets immediately return and then encounter a null check, so no matter what - this patch is either badly formed or logically incorrect.  Changes are required.